pkg/kube/inject/webhook.go

```diff
@@ -662,20 +689,35 @@ func reorderPod(pod *corev1.Pod, req InjectionParameters) error {
 		proxyLocation = MoveFirst
 	}
 
-	// Proxy container should be last, unless HoldApplicationUntilProxyStarts is set
-	// This is to ensure `kubectl exec` and similar commands continue to default to the user's container
-	pod.Spec.Containers = modifyContainers(pod.Spec.Containers, ProxyContainerName, proxyLocation)
-	// Validation container must be first to block any user containers
-	pod.Spec.InitContainers = modifyContainers(pod.Spec.InitContainers, ValidationContainerName, MoveFirst)
-	// Init container must be last to allow any traffic to pass before iptables is setup
-	pod.Spec.InitContainers = modifyContainers(pod.Spec.InitContainers, InitContainerName, MoveLast)
-	pod.Spec.InitContainers = modifyContainers(pod.Spec.InitContainers, EnableCoreDumpName, MoveLast)
+	if features.EnableNativeSidecars.Get() {
```

Comment:
There is one thing majorly broken by this, that we use injection for Gateways and do the rewrite for them.

Options:
1. We somehow indicate per-template whether it is a "sidecar" or not
2. We actually make the template add it as an `initContainer`. this isn't so bad in the template but makes some other logic more complex

I think we should probably do (2)...